feature/agenda/presentation/src/main/java/com/sergius/agenda/presentation/AgendaViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.sergius.agenda.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.Month
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.minus
+import kotlinx.datetime.plus
+import kotlinx.datetime.todayIn
+import kotlin.time.Clock
+import kotlin.time.ExperimentalTime
+
+@OptIn(ExperimentalTime::class)
+class AgendaViewModel : ViewModel() {
+
+    private val _state = MutableStateFlow(
+        AgendaState(
+            month = getCurrentMonth(),
+            days = getDaysOfMonth()
+        )
+    )
+    val state = _state
+        .onStart {
+            getDaysOfMonth()
```

Comment:
Something to be careful with here: With every new call of onStart (screen rotation, navigation, coming from the foreground, etc.) you add a new independent flow collector and the old one stays active. If you put a println in onEach you‚Äôll likely see multiple of them when resubscribing to the Flow.
This is why it‚Äôs important to add a flag in onStart to check if initial data has been loaded already.
```kotlin
var isInitialized = false
 val state = _state
        .onStart {
            if (isInitialized) return@onStart
            getDaysOfMonth()
            isInitialized = true
        }
```